src/main/java/org/candlepin/resource/EnvironmentResource.java

```diff
@@ -145,21 +155,66 @@ public void deleteEnvironment(@Verify(Environment.class) String envId) {
             throw new NotFoundException(i18n.tr("No such environment: {0}", envId));
         }
 
-        CandlepinQuery<Consumer> consumers = this.envCurator.getEnvironmentConsumers(environment);
+        List<Consumer> consumers = this.envCurator.getEnvironmentConsumers(environment).list();
+        deleteConsumers(environment, consumers);
+
+        log.info("Deleting environment: {}", environment);
+        this.envCurator.delete(environment);
+    }
 
+    private void deleteConsumers(Environment environment, List<Consumer> consumers) {
+        if (consumers.isEmpty()) {
+            log.info("No consumers found in environment: {}", environment);
+            return;
+        }
         // Cleanup all consumers and their entitlements:
         log.info("Deleting consumers in environment {}", environment);
-        for (Consumer c : consumers.list()) {
+        for (Consumer c : consumers) {
```

Comment:
If we're already iterating over each consumer, it seems like it'd be best to pull the necessary cert IDs here, rather than doing multiple individual iterations for each type and serial.

Reply:
Done.